src/main/java/com/aws/greengrass/localdebugconsole/DashboardServer.java

```diff
@@ -212,6 +238,102 @@ public void onMessage(WebSocket conn, String msg) {
+                        } else {
+                            pubSubWatchList.get(conn).computeIfAbsent(subId, (a) -> {
+                                Consumer<PublishEvent> cb = (c) -> {
+                                    CommunicationMessage resMessage =
+                                            new CommunicationMessage(subId, topicFilter, c.getTopic(),
+                                                    new String(c.getPayload()));
+                                    sendIfOpen(conn, new Message(MessageType.PUB_SUB_MSG, resMessage));
+                                };
+                                SubscribeRequest subReq = SubscribeRequest.builder().callback(cb)
+                                        .receiveMode(ReceiveMode.RECEIVE_ALL_MESSAGES).topic(topicFilter)
+                                        .serviceName(SERVICE_NAME).build();
+                                pubSubIPCAgent.subscribe(subReq);
+                                return subReq;
+                            });
+                        }
+                        sendIfOpen(conn, new Message(MessageType.RESPONSE, packedRequest.requestID, true));
+                    } catch (Exception e) {
+                        sendIfOpen(conn, new Message(MessageType.RESPONSE, packedRequest.requestID, e.getMessage()));
+                    }
+                    break;
+                }
+                case publishToPubSubTopic: {
+                    JsonNode tree;
+                    try {
+                        tree = jsonMapper.readTree(req.args[0]);
+                    } catch (JsonProcessingException e) {
+                        sendIfOpen(conn, new Message(MessageType.RESPONSE, packedRequest.requestID, e.getMessage()));
+                        break;
+                    }
+
+                    String topic = tree.get("topic").textValue();
+                    String destination = tree.get("destination").textValue();
+                    String payload = tree.get("payload").textValue();
+                    try {
+                        if ("iotcore".equals(destination)) {
+                            mqttClient.publish(Publish.builder()
+                                    .topic(topic)
+                                    .payload(payload.getBytes())
+                                    .build());
+                        } else {
+                            pubSubIPCAgent.publish(topic, payload.getBytes(), SERVICE_NAME);
+                        }
+                        sendIfOpen(conn, new Message(MessageType.RESPONSE, packedRequest.requestID, true));
+                    } catch (Exception e) {
+                        sendIfOpen(conn, new Message(MessageType.RESPONSE, packedRequest.requestID, e.getMessage()));
+                    }
+                    break;
+                }
+                case unsubscribeToPubSubTopic: {
+                    SubscribeRequest subReq = pubSubWatchList.get(conn).remove(req.args[0]);
```

Comment:
If there was no subscription already, then I don't think we care. And yes it will have a value, and even if it doesn't we'll log out an error, not crash.